compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -100,21 +100,63 @@ bool isAArch64(IREE::HAL::ExecutableVariantOp variantOp) {
   return triple && triple.getValue().isAArch64();
 }
 
+static Optional<IREE::HAL::ExecutableVariantOp> getParentExecutableVariantOp(
```

Comment:
Isnt this same as `getEntryPoint`?

Reply:
Ah, I missed that one! They seem different. This one is returning an `ExecutableVarianOp`, which is the one with the target information. `getEntryPoint` returns an `ExecutableExportOp`. I think we should keep both but I can align the new one with the implementation with `getEntryPoint`.

Reply:
For this particular case, yes. Sure, I'll align the implementations.

Reply:
I used `FailureOf<...>` but it feels a bit weird (perhaps it's just me) to check `succeeded(variableOp)` to know if `variableOp` actually has a value. Perhaps we should use `Optional` instead but I won't change it as part of this PR since changes would span to other files.